vcl_templates/www.vcl.erb

```diff
@@ -184,9 +184,20 @@ sub vcl_recv {
 
   # FIXME: Prefer a fallback director if we move to Varnish 3
   if (req.restarts > 2) {
-    set req.backend = F_mirror1;
-    set req.http.host = "<%= config.fetch('provider1_mirror_hostname') %>";
-    set req.http.Fastly-Backend-Name = "mirror1";
+    set req.backend = F_mirrorS3;
+    set req.http.host = "<%= config.fetch('s3_mirror_hostname') %>";
+    set req.http.Fastly-Backend-Name = "mirrorS3";
+
+    if (req.url ~ "^(/.*)?/([^#?\s]+)\.(atom|css|csv|docx|gif|html|ico|ics|jpeg|jpg|js|json|ods|odt|pdf|png)([\?#]+.*)?$") {
```

Comment:
We have URLs that contain a dot in the filename, so we need to differentiate when it's an extension and when it's not. The list of extensions will need to be reviewed, and I agree it's fragile, but unless we force people not to publish filenames that contain a dot I don't know what else we can do

The s3 prefix needs to be added to all the requests, that is the path where the files are stored in the S3 bucket